discovery/azure/azure.go

```diff
@@ -531,6 +535,10 @@ func mapFromVM(vm compute.VirtualMachine) virtualMachine {
 		computerName = *(vm.VirtualMachineProperties.OsProfile.ComputerName)
 	}
 
+	if vm.VirtualMachineProperties != nil && vm.VirtualMachineProperties.HardwareProfile != nil {
+		size = string(vm.VirtualMachineProperties.HardwareProfile.VMSize)
+	}
```

Comment:
For the sake of simplicity, do you mind breaking out the `if vm.VirtualMachineProperties != nil` above this block and moving this block into that, so it would be like:

```
if vm.VirtualMachineProperties != nil {
 if vm.VirtualMachineScaleSetVMProperties.OsProfile != nil && vm.VirtualMachineProperties.OsProfile.ComputerName != nil {
  ...
 }
 if vm.VirtualMachineProperties.HardwareProfile != nil {
  ...
 }
}